compose-builder/gen_secure_compose_ext.sh

```diff
@@ -50,7 +50,7 @@ ADD_SERVICE_SECURE_FILE_TEMPLATE="add-service-secure-template.yml"
 SERVICE_EXT_COMPOSE_PATH=./"$GEN_EXT_DIR"/add-"$service_name"-secure.yml
 sed 's/${SERVICE_NAME}:/'"$service_name"':/g' "$ADD_SERVICE_SECURE_FILE_TEMPLATE" > "$SERVICE_EXT_COMPOSE_PATH"
 sed -i 's/${SERVICE_KEY}/'"$service_key"'/g' "$SERVICE_EXT_COMPOSE_PATH"
-sed -i 's/${EXECUTABLE}/'"$executable"'/g' "$SERVICE_EXT_COMPOSE_PATH"
+sed -i 's,${EXECUTABLE},'"$executable"',g' "$SERVICE_EXT_COMPOSE_PATH"
```

Comment:
Why is this change needed? How does it impact the generated compose file?

Reply:
If the executable has a slash in it, such as device-bacnet-ip/device-bacnet-c, then the above command without the fix, is interpeted as

```
sed -i 's/${EXECUTABLE}/'device-bacnet-ip/device-bacnet-c/g' "$SERVICE_EXT_COMPOSE_PATH"
```

This is illegal because the sed syntax is

s (delimeter) (search-string) (delimeter) (replace-string) (delimeter) (options)

This update changes the delimiter to a comma to as to not conflict with the path separator.

While comma is an allowed path character in Unix, I have never seen anyone use a comma in their paths, but I see slash used all the time.

Reply:
This script will be called during `make build`
Have you run `make build` to verify no un-foreseen changes to the generated compose file.

Reply:
> If the executable has a slash in it, such as device-bacnet-ip/device-bacnet-c, then the above command without the fix, is interpreted as
> 
> ```
> sed -i 's/${EXECUTABLE}/device-bacnet-ip/device-bacnet-c/g' "$SERVICE_EXT_COMPOSE_PATH"
> ```
> 
> This is illegal because the sed syntax is
> 
> s (delimeter) (search-string) (delimeter) (replace-string) (delimeter) (options)
> 
> This update changes the delimiter to a comma to as to not conflict with the path separator.
> 
> While comma is an allowed path character in Unix, I have never seen anyone use a comma in their paths, but I see slash used all the time.

just a bit picky here: there is a potential issue if there is query string with comma (,) as part of URL path; but so far I haven't seen any in EdgeX, so we should be fine here.

Reply:
"make build" results in no changes to the generate compose files.